src/gateway_config_miner.erl

```diff
@@ -0,0 +1,106 @@
+-module(gateway_config_miner).
+
+-behavior(gen_server).
+
+-include("gateway_config.hrl").
+-define(SERVICE, 'helium.local.api').
+
+%% gen_server
+-export([
+    start_link/0,
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2
+]).
+
+%% api
+-export([
+    status/0,
+    pubkey/0,
+    add_gateway/3
+]).
+
+-record(state, {
+    connection = undefined :: grpc_client:connection() | undefined
+}).
+
+-spec status() -> {ok, [tuple()]} | {error, term()}.
+status() ->
+    gen_server:call(?MODULE, status).
+
+-spec pubkey() -> {ok, {PubKey :: binary(), OnboardingKey :: binary()}} | {error, term()}.
+pubkey() ->
+    gen_server:call(?MODULE, pubkey).
+
+-spec add_gateway(
+    Owner :: libp2p_crypto:pubkey_bin(),
+    Payer :: libp2p_crypto:pubkey_bin(),
+    Mode :: full | dataonly | light
+) -> {ok, binary()} | {error, term()}.
+add_gateway(Owner, Payer, Mode) ->
+    gen_server:call(?MODULE, {add_gateway, [Owner, Payer, Mode]}).
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+init(_) ->
+    {ok, #state{}}.
+
+handle_call(Call, From, State = #state{connection = undefined}) ->
+    GrpcPort = application:get_env(gateway_config, grpc_port, 4467),
+    case grpc_client:connect(tcp, "localhost", GrpcPort) of
+        {ok, Connection = #{http_connection := Pid}} ->
+            erlang:monitor(process, Pid),
+            handle_call(Call, From, State#state{connection = Connection});
+        {error, Error} ->
+            {reply, {error, Error}, State}
+    end;
+handle_call(status, _From, State = #state{connection = Connection}) ->
+    case call_unary(Connection, height, #{}) of
+        {ok, #{result := #{height := Height}}} ->
+            {reply, {ok, [{"connected", "yes"}, {"height", integer_to_list(Height)}]}, State};
+        {error, _} ->
+            {reply, {ok, [{"connected", "no"}]}, State#state{connection = undefined}}
```

Comment:
i would have thought this would leave the old connection dangling but in testing i see an interruption to the http connection on the other side _seems_ to reliably kill the http_conn pid here which gets routed correctly to the `DOWN` monitor handler.